File: Medium/1006-Clumsy Factorial/Mathematical.rs
impl Solution {
    pub fn clumsy(n: i32) -> i32 {
        match n % 4 {
            0 => (n + 1).max(7),
            1 | 2 if n < 4 => n,
            1 | 2 => n + 2,
            _ => (n - 1).max(6),
        }
    }
}

File: Medium/1006-Clumsy Factorial/Mathematical.rb
# @param {Integer} n
# @return {Integer}
def clumsy(n)
    if n % 4 == 0
        return [n + 1, 7].max
    elsif n < 3
        return n
    elsif n % 4 == 1 or n % 4 == 2
        return n + 2
    else
        return [n - 1, 6].max
    end
end
